File: backend/src/main/java/com/edu/management/entity/Campus.java
package com.edu.management.entity;

import com.edu.management.enums.CampusStatus;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "campuses")
public class Campus {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 100)
    private String name;
    
    @Column(length = 255)
    private String address;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private CampusStatus status = CampusStatus.ACTIVE;
    
    @CreationTimestamp
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: mobile-app/src/api/campus.ts
import { get } from '../utils/request'

export interface Campus {
  id: number
  name: string
  address: string
  status: 'ACTIVE' | 'INACTIVE'
}

// 获取所有校区
export const getAllCampuses = () => {
  return get<Campus[]>('/campuses')
}

// 获取活跃校区
export const getActiveCampuses = () => {
  return get<Campus[]>('/campuses/active')
}
